agent/acs/handler/acs_handler_test.go

```diff
@@ -202,13 +204,14 @@ func TestHandlerReconnectsOnConnectErrors(t *testing.T) {
 	mockWsClient.EXPECT().Close().Return(nil).AnyTimes()
 	gomock.InOrder(
 		// Connect fails 10 times
-		mockWsClient.EXPECT().Connect().Return(io.EOF).Times(10),
+		mockWsClient.EXPECT().Connect(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, io.EOF).Times(10),
 		// Cancel trying to connect to ACS on the 11th attempt
 		// Failure to retry on Connect() errors should cause the
 		// test to time out as the context is never cancelled
-		mockWsClient.EXPECT().Connect().Do(func() {
+
+		mockWsClient.EXPECT().Connect(gomock.Any(), gomock.Any(), gomock.Any()).Do(func(interface{}, interface{}, interface{}) {
 			cancel()
-		}).Return(nil).MinTimes(1),
+		}).Return(nil, nil).MinTimes(1),
```

Comment:
The first `nil` here should be some sort of mock or dummy disconnect timer, right? (this comment applies to other lines of this file where the same change is made as well)

Reply:
Yes, you are right. This is the test due to which I added the nil [check.](https://github.com/aws/amazon-ecs-agent/pull/3751#discussion_r1303570546)
Let me create a timer in this mock to return.